### PR TITLE
Run pyupgrade --py36-plus

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: mypy/checker.py [hunk 1]
```diff
@@ -92,22 +92,15 @@
...
 # and methods of classes not defined within a function can be deferred.
-DeferredNode = NamedTuple(
-    'DeferredNode',
-    [
-        ('node', DeferredNodeType),
-        ('active_typeinfo', Optional[TypeInfo]),  # And its TypeInfo (for semantic analysis
-                                                  # self type handling)
-    ])
+class DeferredNode(NamedTuple):
+    node: DeferredNodeType
```
Comment (JelleZijlstra) [inline mypy/checker.py:96]: It ate the comments

File: mypy/sitepkgs.py [hunk 2]
```diff
@@ -1,4 +1,3 @@
-from __future__ import print_function
```
Comment (ethanhs) [inline mypy/sitepkgs.py:1]: This file "needs" to run on Python 2 for Python 2 support (whatever that is as of now), so this import should probably stay.

**Comment by ethanhs** [comment]
Perhaps this could be done one file at a time to make it easier to review?

**Comment by hauntsaninja** [comment]
Yeah, agreed. Seems like this needs a close-ish review + would cause many more merge conflicts, so breaking it up over files and time sounds good.

**Pull request closed without merging**